src/components/calcite-card/calcite-card.tsx

```diff
@@ -0,0 +1,193 @@
+import { ENTER, SPACE } from "../../utils/keys";
+import { getElementDir } from "../../utils/dom";
+import { VNode } from "@stencil/core/dist/declarations";
+
+/**
+ * @slot thumbnail - [Required] A slot for adding a thumnail to the card.
+ * @slot header - [Required] A slot for adding a heading and an icon to the card.
+ * @slot - A slot for adding subheader/description content.
+ * @slot action - A slot for adding a single action as a button.
+ * @slot footer-leading - A slot for adding a leading footer.
+ * @slot footer-trailing - A slot for adding a trailing footer.
+ */
+
+@Component({
+  tag: "calcite-card",
+  styleUrl: "calcite-card.scss",
+  shadow: true
+})
+export class CalciteCard {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteCardElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**  true, the card can't be clicked and is visually muted.  */
+  @Prop({ reflect: true }) disabled?: boolean = false;
+
+  /**  When true, the cards content is waiting to be loaded. This state shows a busy indicator.*/
+  @Prop({ reflect: true }) loading?: boolean = false;
+
+  /**
+   * Indicates whether the image's height is respected for the crop of the thumbnail.
+   * When false (default), the image fills the whole thumbnail space.
+   * When true, the height of the thumbnail is used to crop and may not fill the container for the thumbnail.
+   */
+  @Prop({ reflect: true }) respectImageHeight?: boolean = false;
+
+  /** Indicates whether the card is selected. */
+  @Prop({ reflect: true, mutable: true }) selected?: boolean = false;
+
+  /** Indicates whether the card is selected. */
+  @Prop({ reflect: true, mutable: true }) selectable?: boolean = false;
+
+  /**  The theme of the card.*/
+  @Prop({ reflect: true, mutable: true }) theme: "light" | "dark" = "light";
+
+  /**  The theme of the card.*/
+  @Prop({ reflect: true, mutable: true }) appearance: "wide" | "default" =
+    "wide";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /** Fired when a selectable card is selected */
+  @Event() calciteCardSelected: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback() {
+    let themes = ["dark", "light"];
+    if (!themes.includes(this.theme)) this.theme = "light";
+  }
+
+  render() {
+    const dir = getElementDir(this.el);
+    return (
+      <Host dir={dir}>
+        <a>
```

Comment:
This empty anchor tag seems odd. Whats the reasoning for it? Could it be accomplished with a role?

Reply:
Yeah, that's a placeholder. We need a way to allow folks to:

- make the whole card clickable (i'd imagine a href passed to that currently empty `a` tag)
- make just the image clickable, accepting a href
- do not allow the card to be clickable, just relying on any slotted content to have links itself

So - one would wrap the card in an anchor, one would just wrap the image in an anchor, one wouldn't do either. Mostly wanted to get thoughts on that.

Reply:
> make the whole card clickable

Can't they just add an `onClick` to the card? Do we need to make it an anchor?

> make just the image clickable, accepting a href

They should be able to do this already, they can use the `thumbnail` slot to pass an `anchor` surrounding an `img`.

Reply:
Cool, thumbnail makes sense, and I guess folks could add their own onClick but, IMO we'd end up with a lot of usability issues with cards not being "right click to open in new tab" etc for cases where they just link to detail items...

I could see `calcite-card` accepting a href attribute, and if that's present - wrap in an anchor, if not, wrap in a div?

Reply:
> I could see calcite-card accepting a href attribute, and if that's present - wrap in an anchor, if not, wrap in a div?

I would say they should wrap the card in an anchor in that case. Otherwise, we have to accept all the anchor attributes like target, href, hreflang, rel, refererpolicy... etc

Thoughts?

Reply:
Hmm, yeah good point that's kind of getting into that "spreading attributes to a child element through whitelist / blacklist"  - I think having them wrap in an anchor of their own is reasonable as long as it doesn't get in the way of focus or hover / active styling or focus order...

Reply:
Yeah, the only good way of doing this is through a slot but seems like them wrapping the component would be better.